src/pages/AppConfigure.vue

```diff
@@ -35,14 +42,31 @@ async function handleSave() {
 
 async function updateSmartContractTransactions(app: AppConfig) {
   try {
-    store.commit('showLoader', 'Updating app name in smart contract...')
-    await setAppName(app.name)
+    if (app.name !== currentConfig.name) {
+      store.commit('showLoader', 'Updating app name in smart contract...')
```

Comment:
Check if you need to hide the loader either after each step or in a `finally { store.commit("hideLoader") }`.